source/crc.c

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include "checksums.h"
+
+#include "aws/checksums/crc.h"
+#include "aws/common/byte_buf.h"
+
+PyObject *aws_py_checksums_crc32(PyObject *self, PyObject *args) {
+    (void)self;
+    // (void)args;
+
+    Py_buffer input_buf;
+    PyObject *py_previousCrc32;
+
+    if (!PyArg_ParseTuple(args, "s*O", &input_buf, &py_previousCrc32)) {
+        return PyErr_AwsLastError();
+    }
+
+    uint32_t previousCrc32 = PyLong_AsUnsignedLong(py_previousCrc32);
+
+    if (previousCrc32 == (uint32_t)-1 && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    struct aws_byte_buf input = aws_byte_buf_from_array(input_buf.buf, input_buf.len);
```

Comment:
Trivial: Are these conversions to aws_byte_buf redundant?  (It looks like it just copies the pointer/len directly into the byte buffer structure, and then we re-feed the same values into aws_checksums_crc32)

Reply:
corrected